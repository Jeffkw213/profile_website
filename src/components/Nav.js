import React from 'react'
//This is gonna be the navigation bar
//Need to create a list of 
const Item = (props) =>(
    <lu>
        <li>
            <a href={props.item.url}>{props.item.title}</a>
        </li>
        <li>{props.item.author}</li>
        <li>{props.item.num_comments}</li>
        <li>{props.item.points}</li>
    </lu>
)

export default function Nav(props) {
  return (
    <div>
      
    </div>
  )
}
