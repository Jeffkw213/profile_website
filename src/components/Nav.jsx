//This is gonna be the navigation bar
//Need to create a list of

export const List = (props) => (
  <ul>
    <li>
      <a href={props.Linkedin} target="_blank" rel="noopener noreferrer">
        Linkedin
      </a>
    </li>
    <li>
      <a href={props.Github} target="_blank" rel="noopener noreferrer">
        Github
      </a>
    </li>
    <li>
      <a href="/Resume.pdf" target="_blank">
        Resume
      </a>
    </li>
  </ul>
);

export default function Nav(props) {
  return (
    <div>
      <List {...props} />
    </div>
  );
}
