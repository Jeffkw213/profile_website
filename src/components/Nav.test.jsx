import renderer from 'react-test-renderer';
import Nav, { List } from './Nav';
import rec from '../data/records.json';

function toJson(component) {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result;
}

test('List renders when given data', () => {
  const component = renderer.create(<List {...rec.Social} />);

  let tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test('Nav renders when given data', () => {
  const component = renderer.create(<Nav {...rec.Social} />);

  let tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
