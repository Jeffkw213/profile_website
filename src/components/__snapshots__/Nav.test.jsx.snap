// Vitest Snapshot v1

exports[`List renders when given data 1`] = `
<nav
  className="flex justify-center"
>
  <ul
    className=""
  >
    <li
      className="inline-block"
    >
      <a
        href="https://www.linkedin.com/in/jeff-kwan-881a41203/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Linkedin Logo"
          className="h-16 opacity-[.35]"
          src="/src/data/LI-In-Bug.png"
        />
      </a>
    </li>
    <li
      className="inline-block"
    >
      <a
        href="https://github.com/Jeffkw213"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Git Logo"
          className="h-16 opacity-[.35]"
          src="/src/data/GitHub-Mark-64px.png"
        />
      </a>
    </li>
    <li
      className="inline-block ml-2"
    >
      <a
        href="mailto:jeffkwan10@gmail.com?subject=Hi%20I'm%20Stupid"
      >
        <img
          alt="Gmail Logo"
          className="h-16 opacity-[.35]"
          src="/src/data/Gmail_Logo_64px.png"
        />
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Nav renders when given data 1`] = `
<div
  className="h-20 relative w-full py-2 text-center bg-black"
  id="contacts"
>
  <nav
    className="flex justify-center"
  >
    <ul
      className=""
    >
      <li
        className="inline-block"
      >
        <a
          href="https://www.linkedin.com/in/jeff-kwan-881a41203/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Linkedin Logo"
            className="h-16 opacity-[.35]"
            src="/src/data/LI-In-Bug.png"
          />
        </a>
      </li>
      <li
        className="inline-block"
      >
        <a
          href="https://github.com/Jeffkw213"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Git Logo"
            className="h-16 opacity-[.35]"
            src="/src/data/GitHub-Mark-64px.png"
          />
        </a>
      </li>
      <li
        className="inline-block ml-2"
      >
        <a
          href="mailto:jeffkwan10@gmail.com?subject=Hi%20I'm%20Stupid"
        >
          <img
            alt="Gmail Logo"
            className="h-16 opacity-[.35]"
            src="/src/data/Gmail_Logo_64px.png"
          />
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
